### PR TITLE
fix: diffs always occurring when multiple columns exist

### LINKED ISSUE #2054: Conditional masking policy with multiple columns always forces replacement
**Provider Version**

0.70.1

**Terraform Version**

1.5.4

**Describe the bug**

We are creating a [conditional masking policy](https://docs.snowflake.com/en/sql-reference/sql/create-masking-policy#example-conditional-masking-policy) with multiple columns in the signature. The masking policy is created correctly in Snowflake, but when running `terraform plan` afterwards it always claims that the `signature` has changed which forces replacement.

**Expected behavior**

The masking policy should not change.

**Code samples and commands**

Example masking policy that fails:
```
resource "snowflake_masking_policy" "example_conditional_masking_policy" {
  name = "EXAMPLE_CONDITIONAL_MASKING_POLICY"

  database           = "EXAMPLE_DB"
  schema             = "EXAMPLE_SCHEMA"
  return_data_type   = "NUMBER"
  masking_expression = <<-EOF
  case
    when current_role() in ('ADMIN') then val
    when customer_id <= 0 then 0
    else val
  end
EOF
  signature {
    column {
      name = "VAL"
      type = "NUMBER"
    }
    column {
      name = "CUSTOMER_ID"
      type = "NUMBER"
    }
  }
}
```

Output from `terraform plan`:
```
 # snowflake_masking_policy.example_conditional_masking_policy must be replaced
-/+ resource "snowflake_masking_policy" "example_conditional_masking_policy" {
      ~ id                    = "EXAMPLE_DB|EXAMPLE_SCHEMA|EXAMPLE_CONDITIONAL_MASKING_POLICY" -> (known after apply)
      ~ masking_expression    = <<-EOT
            case
              when current_role() in ('ADMIN') then val
              when customer_id <= 0 then 0
              else val
            end
        EOT
        name                  = "EXAMPLE_CONDITIONAL_MASKING_POLICY"
      ~ qualified_name        = "\"EXAMPLE_DB\".\"EXAMPLE_SCHEMA\".\"EXAMPLE_CONDITIONAL_MASKING_POLICY\"" -> (known after apply)
        # (4 unchanged attributes hidden)

      ~ signature {
          + column {
              + name = "CUSTOMER_ID"
              + type = "NUMBER" # forces replacement
            }

            # (1 unchanged block hidden)
        }
      - signature {
          - column {
              - name = "CUSTOMER_ID" -> null
              - type = "NUMBER" -> null # forces replacement
            }
        }
    }
```

**Additional context**

It looks like it also marks the masking expression as changed - could be related to issues with heredocs in https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/2053?

### PR DESCRIPTION
**Pull request by YAhiru**
ReadMaskingPolicy function increased signature by the number of columns when there were multiple columns in the Masking Policy.
Therefore, a masking policy with multiple columns was always in a state where differences occurred.

## Test Plan

* [x] acceptance tests

## References

* fixes #2054

### CODE REVIEW

**Comment by sfc-gh-asawicki** [comment]
Hey @YAhiru. Thanks for the pull request! We will review it tomorrow (cc: @sfc-gh-jcieslak).

File: pkg/resources/masking_policy_acceptance_test.go [hunk 1]
```diff
@@ -118,3 +118,57 @@ func maskingPolicyConfigMultiline(n string, name string, databaseName string, sc
...
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		PreCheck:     func() { acc.TestAccPreCheck(t) },
+		CheckDestroy: nil,
+		Steps: []resource.TestStep{
+			{
```
Comment (sfc-gh-jcieslak) [inline pkg/resources/masking_policy_acceptance_test.go:133]: Could you add
```go
ConfigPlanChecks: resource.ConfigPlanChecks{
	PostApplyPostRefresh: []plancheck.PlanCheck{
		plancheck.ExpectEmptyPlan(),
	},
},
```
I think the plan is checked implicitly in the testing framework, but please add it, just to be explicit and sure that no plan is produced after the initial apply.
Other than that everything looks good. Thanks for the contribution! :)

Comment (YAhiru) [reply pkg/resources/masking_policy_acceptance_test.go:134]: @sfc-gh-jcieslak 
Thanks for the review!
I added :)
https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/2686/commits/4d8a779af7c3f8ca59c3b4dc205f0724fe26bf6a

Comment (sfc-gh-jcieslak) [reply pkg/resources/masking_policy_acceptance_test.go:134]: Thank you :)

**Review by sfc-gh-jcieslak — APPROVED** [review]

**Pull request merged**